datapackage.js

```diff
@@ -15,14 +16,31 @@ exports.exportDatapackage = function() {
     datapackage = null;
   });
 
-  ipc.once('sendDatapackage', function(e, data) {
-    var data = datapackageJson(data)
+  ipc.once('sendDatapackage', function(e, data, includeHeaders) {
+    // data here is the object generated by filling out the form
+    //debugger;
+    // parse available globals to get suggested names for the data package
+    var currentFileName = window.getTitle();
+    var suggestedFileName = path.basename(currentFileName).replace('.csv', '');
+
+    // issue calls to get the JSON rendering of the header row
+    if(includeHeaders === "true"){
+      window.webContents.send('schemaFromHeaders');
+      ipc.once('jsonHeaders', function(event, json){
+        //assign the headers to the JSON datapackage
+        // for some reason the JSON becomes inverted at this point
+        data["fields"] = json["fields"];
```

Comment:
The schema should be contained in the resources - see http://data.okfn.org/doc/data-package#-a-href-https-github-com-datasets-gdp-world-gdp-a- for and example :smile: